language_service/src/hover.rs

```diff
@@ -25,67 +29,458 @@ pub(crate) fn get_hover(
     offset: u32,
 ) -> Option<Hover> {
     // Map the file offset into a SourceMap offset
-    let offset = map_offset(&compilation.source_map, source_name, offset);
-    let package = &compilation.package;
+    let offset = map_offset(&compilation.unit.sources, source_name, offset);
+    let package = &compilation.unit.ast.package;
 
-    let mut callable_finder = CallableFinder {
+    let mut hover_visitor = HoverVisitor {
+        compilation,
         offset,
         header: None,
         start: 0,
         end: 0,
     };
-    callable_finder.visit_package(package);
 
-    callable_finder.header.map(|header| Hover {
+    hover_visitor.visit_package(package);
+
+    hover_visitor.header.map(|header| Hover {
         contents: header,
         span: Span {
-            start: callable_finder.start,
-            end: callable_finder.end,
+            start: hover_visitor.start,
+            end: hover_visitor.end,
         },
     })
 }
 
-struct CallableFinder {
+struct HoverVisitor<'a> {
+    compilation: &'a Compilation,
     offset: u32,
     header: Option<String>,
```

Comment:
Suggest renaming to `contents` to match [LSP spec](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#hover)

Reply:
done